book/src/design/proving-system/vanishing.md

```diff
@@ -0,0 +1,79 @@
+# Vanishing argument
+
+Having committed to the circuit assignments, the prover now needs to demonstrate that the
+various circuit relations are satisfied:
+
+- The custom gates, represented by polynomials $\text{gate}_i(X)$.
+- The rules of the lookup arguments.
+- The rules of the equality constraint permutations.
+
+Each of these relations is inherently represented as a polynomial of degree $d$ (the
```

Comment:
```suggestion
Each of these relations is represented as a polynomial of degree $d$ (the
```

Reply:
Hmm, I was trying to distinguish between the relation itself (which has degree `d` with respect to the columns), and the relation polynomial (which has degree `d(n-1)` with respect to `X`). Suggestions on alternative wording?

Reply:
Actually, I could just say that ☺️